src/bin/pgcopydb/ld_stream.c

```diff
@@ -35,6 +35,150 @@
 static bool updateStreamCounters(StreamContext *context,
 								 LogicalMessageMetadata *metadata);
 
+/*
+ * Determines whether a message needs to be filtered out based on the
+ * provided filters. The caller function should have already parsed and
+ * validated the filters.
+ */
+bool
+ShouldFilterOutMessage(SourceFilters *filters, char *nspname, char *relname)
+{
+	/*
+	 * Validate nspname is not NULL or empty
+	 */
+	if (nspname == NULL || IS_EMPTY_STRING_BUFFER(nspname))
+	{
+		log_error("BUG: nspname is NULL or empty");
+		return false;
+	}
+
+	/*
+	 * Validate relname is not NULL or empty
+	 */
+	if (relname == NULL || IS_EMPTY_STRING_BUFFER(relname))
+	{
+		log_error("BUG: relname is NULL or empty");
+		return false;
+	}
+
+	/*
+	 * Filter-out pgcopydb.* in any case
+	 */
+	if (streq(nspname, "pgcopydb"))
+	{
+		log_info("Filtering out message for pgcopydb internal tables \"%s\"", nspname);
+		return true;
+	}
+
+	/*
+	 * If no filters are set, then we don't need to filter out any messages
+	 */
+	if (filters->type == SOURCE_FILTER_TYPE_NONE)
+	{
+		return false;
+	}
```

Comment:
We can't use the return value of the function for two separate meanings: either it means the function could operate as intended (see the previous return false cases) or it means that `ShouldFilterOutMessage` is true/false. I advice doing like we do in most of the other code where that happens, return true/false to indicate that the function could operate as intended, and add a `bool *shouldFitlterOutMessage` argument to the function.

Reply:
[9250199](https://github.com/dimitri/pgcopydb/pull/614/commits/925019998251e5634944189a9975af6e75c8172a)